extensions/kafka-connect/src/test/java/com/hazelcast/jet/kafka/connect/KafkaConnectRandomIntIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.kafka.connect;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.jet.pipeline.Sinks;
+import com.hazelcast.jet.pipeline.StreamStage;
+import com.hazelcast.jet.pipeline.test.AssertionCompletedException;
+import com.hazelcast.jet.pipeline.test.AssertionSinks;
+import org.apache.kafka.connect.data.Values;
+import org.junit.Test;
+
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.CompletionException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class KafkaConnectRandomIntIntegrationTest extends JetTestSupport {
+
+    public static final int ITEM_COUNT = 10_000;
+
+    @Test
+    public void readFromRandomSource() throws Exception {
+        System.setProperty("hazelcast.logging.type", "log4j2");
+        Properties randomProperties = new Properties();
+        randomProperties.setProperty("name", "random-source-connector");
+        randomProperties.setProperty("connector.class", "sasakitoa.kafka.connect.random.RandomSourceConnector");
+        randomProperties.setProperty("generator.class", "sasakitoa.kafka.connect.random.generator.RandomInt");
+        randomProperties.setProperty("tasks.max", "1");
+        randomProperties.setProperty("messages.per.second", "1000");
+        randomProperties.setProperty("topic", "test");
+        randomProperties.setProperty("task.summary.enable", "true");
+
+        Pipeline pipeline = Pipeline.create();
+        StreamStage<String> streamStage = pipeline.readFrom(KafkaConnectSources.connect(randomProperties))
+                .withoutTimestamps()
+                .map(record -> Values.convertToString(record.valueSchema(), record.value()));
+        streamStage.writeTo(Sinks.logger());
+        streamStage
+                .writeTo(AssertionSinks.assertCollectedEventually(60,
+                        list -> assertEquals(ITEM_COUNT, list.size())));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.addJar(Objects.requireNonNull(this.getClass()
+                        .getClassLoader()
+                        .getResource("random-connector-1.0-SNAPSHOT.jar"))
+                .getPath()
+        );
+
+        Config config = smallInstanceConfig();
+        config.getJetConfig().setResourceUploadEnabled(true);
+        Job job = createHazelcastInstance(config).getJet().newJob(pipeline, jobConfig);
+
+        sleepAtLeastSeconds(5);
```

Comment:
Is there any reason to wait here?

Reply:
it seems there's none: https://github.com/hazelcast/hazelcast/pull/23312/commits/49b5282b0ce857fdca8d26ab40165bf57baf8506